.circleci/config.yml.in

```diff
@@ -99,6 +99,81 @@ commands:
       - brew_install:
           formulae: libtool
 
+  apt_install:
```

Comment:
At first sight this entire blob does not really look like an obvious improvement. It's hard to figure out what it does, or why it's needed. Same for `pip_install` I would say.

We might as well just inline everything I think, there would be a bit of repetition, but at least things would remain obvious. My second-best suggestion would be to revert to what you had before, and rely on `regenerate.py` (although I'm not a fan either).

But I'll leave it up to you to decide.

Reply:
> At first sight this entire blob does not really look like an obvious improvement. It's hard to figure out what it does, or why it's needed.

It handles running as super-user, updating the repository database, confirming choices, and suppressing output for you. Compare

```yaml
steps:
  - run: sudo apt update -qy && sudo apt install -qy $PACKAGES
```

with

```yaml
steps:
  - apt_install:
      args: $PACKAGES
```

Not having this as a central command means that every call to it will be a little different. In times this means only cluttering the log (forgetting `-q`), whereas in other situations it fails the job (forgetting `sudo`).

> Same for `pip_install` I would say.

Same argument. Compare

```yaml
steps:
  - run: pip install --user --progress-bar=off $PACKAGES
```

with 


```yaml
steps:
  - pip_install:
      args: $PACKAGES
```

Reply:
Hm, a fairer comparison would be to compare the repeated use of 

```
steps:
    - run: sudo apt update -qy && sudo apt install -qy $PACKAGES
```

vs

```
  apt_install:
    parameters:
      args:
        type: string
      descr:
        type: string
        default: ""
      update:
        type: boolean
        default: true
    steps:
      - run:
          name: >
            <<^ parameters.descr >> apt install << parameters.args >> <</ parameters.descr >>
            <<# parameters.descr >> << parameters.descr >>            <</ parameters.descr >>
          command: |
            <<# parameters.update >> sudo apt update -qy  <</ parameters.update >>
            sudo apt install << parameters.args >>
```
(and the associated cost of understanding and maintaining it) plus the repeated use of 

```
steps:
    - apt_install: args: $PACKAGES
```